### PR TITLE
Used FastHash for DebugPacket

### PR DESCRIPTION
**Pull request by ulteq**

### CODE REVIEW

File: source/main/network/Network.cpp [hunk 1]
```diff
@@ -187,16 +187,9 @@ bool CheckError()
...
 {
-    char sha1result[250] = {0};
-    if (buffer)
-    {
-        RoR::CSHA1 sha1;
-        sha1.UpdateHash((uint8_t *)buffer, header->size);
-        sha1.Final();
-        sha1.ReportHash(sha1result, RoR::CSHA1::REPORT_HEX_SHORT);
-    }
+    const char* hash = HashData(buffer, header->size).c_str();
```
[the remark this answers is not in the document]
Comment (ulteq) [reply source/main/network/Network.cpp]: > then let the String go out of scope

Are we?

**Comment by ulteq** [comment]
>  didn't even know we had `DebugPacket()`

I used it a lot during the network revamp.

File: source/main/network/Network.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (only-a-ptr) [reply source/main/network/Network.cpp]: I had to test, and oh yes, we are: http://cpp.sh/8tic6

[the remark this answers is not in the document]
Comment (ulteq) [reply source/main/network/Network.cpp]: :confused:

[the remark this answers is not in the document]
Comment (only-a-ptr) [reply source/main/network/Network.cpp]: > Let's get rid of the entire sprintf? 😕

👍

**Pull request merged**